rust/ql/src/queries/security/CWE-696/BadCtorInitialization.ql

```diff
@@ -40,16 +40,19 @@ class StdCall extends Expr {
 class PathElement = AstNode;
 
 query predicate edges(PathElement pred, PathElement succ) {
-  // starting edge
-  exists(CtorAttr ctor, Function f, StdCall call |
+  // starting edge (`#[ctor]` / `#[dtor]` attribute to call)
+  exists(CtorAttr ctor, Function f |
     f.getAnAttr() = ctor and
-    call.getEnclosingCallable() = f and
-    pred = ctor and // source
-    succ = call // sink
+    pred = ctor and
+    succ.(CallExprBase).getEnclosingCallable() = f
+  )
+  or
+  // transitive edge (call to call)
+  exists(Function f |
+    edges(_, pred) and
+    pred.(CallExprBase).getStaticTarget() = f and
+    succ.(CallExprBase).getEnclosingCallable() = f
```

Comment:
I wonder if it would be easier for users to get a path with an additional edge through `f`? So instead of going `call1 -> call2` it would go `call1 -> function -> call2`?

Reply:
I see what you mean.  I'll see what I can do...

Reply:
Updated.